### PR TITLE
Added test and documentation for template level starter_assets

### PR DESCRIPTION
**Pull request by jmkulwik**
This is a follow-up to a PR that was merged without tests: https://github.com/code-dot-org/code-dot-org/pull/41958 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/controllers/level_starter_assets_controller_test.rb [hunk 1]
```diff
@@ -50,6 +50,40 @@ class LevelStarterAssetsControllerTest < ActionController::TestCase
...
   end
 
+  test 'show: returns template level starter_assets when defined' do
+    uuid_name_1 = "#{SecureRandom.uuid}.png"
+    key_1 = "starter_assets/#{uuid_name_1}"
+    uuid_name_2 = "#{SecureRandom.uuid}.jpg"
+    key_2 = "starter_assets/#{uuid_name_2}"
+    file_objs = [
+      MockS3ObjectSummary.new(key_1, 123, 1.day.ago),
+      MockS3ObjectSummary.new(key_2, 321, 2.days.ago)
```
Comment (bencodeorg) [inline dashboard/test/controllers/level_starter_assets_controller_test.rb]: Do you need this second object?

**Review by bencodeorg — APPROVED** [review]

File: dashboard/test/controllers/level_starter_assets_controller_test.rb (hunk 1 above)
Comment (jmkulwik) [reply dashboard/test/controllers/level_starter_assets_controller_test.rb]: We probably don't technically need to mock it for S3 access, but we should seed the child level with a starter asset to ensure it gets overwritten by the parent level.

Comment (bencodeorg) [reply dashboard/test/controllers/level_starter_assets_controller_test.rb]: Yeah I meant the S3 mock, fine to leave as-is...just wanted to make sure I understood.

Comment (jmkulwik) [reply dashboard/test/controllers/level_starter_assets_controller_test.rb]: Fixed!

**Pull request merged**